src/testcases/ClassesTest.java

```diff
@@ -1,5 +1,455 @@
+        assertEquals("e's field list size is 1", 1, classes.getEntities().get(0).getFields().size());
+
+        assertFalse("False when creating field that already exists", classes.createField("e", "a"));
+        assertEquals("e's field list size is still 1", 1, classes.getEntities().get(0).getFields().size());
+    }
+
+    /** test renameField
+     *
+     */
+    @Test
+    public void testRenameField()
+    {
+        Classes classes = new Classes();
+        classes.createClass("e");
+        classes.createField("e", "a");
+        classes.createField("e", "a2");
+
+        assertFalse("False when renaming Field from non-existent class", classes.renameField("fake", "a", "aa"));
+        assertFalse("False when renaming non-existent Field", classes.renameField("e", "fake", "aa"));
+        assertFalse("False when renaming Field to an existing Field name", classes.renameField("e", "a", "a2"));
+
+        assertTrue("Field 'a' renamed to 'aa'", classes.renameField("e", "a", "aa"));
+        assertTrue("Field list contains 'aa'", classes.getEntities().get(0).getFields().contains("aa"));
+        assertFalse("Field list no longer contains 'a'", classes.getEntities().get(0).getFields().contains("a"));
+        assertEquals("Field list size is still 2", 2, classes.getEntities().get(0).getFields().size());
+    }
+
+    /** test deleteField
+     *
+     */
+    @Test
+    public void testDeleteField()
+    {
+        Classes classes = new Classes();
+        classes.createClass("e");
+        classes.createField("e", "a");
+        classes.createField("e", "a2");
+
+        assertFalse("False when deleting Field from non-existent class", classes.deleteField("fake", "a"));
+        assertFalse("False when deleting non-existent Field", classes.deleteField("e", "fake"));
+
+        assertTrue("Deleted Field 'a'", classes.deleteField("e", "a"));
+        assertFalse("Field list no longer contains 'a'", classes.getEntities().get(0).getFields().contains("a"));
+        assertEquals("Field list size is 1", 1, classes.getEntities().get(0).getFields().size());
+    }
+
+    /** test createMethod
+     *
+     */
+    @Test
+    public void testCreateMethod()
+    {
+        Classes classes = new Classes();
+        classes.createClass("e");
+
+        assertFalse("False when creating method for non-existent class", classes.createMethod("fake", "a"));
+
+        assertTrue("Added method 'a' for class 'e'", classes.createMethod("e", "a"));
+        assertTrue("e's methods list contains 'a'", classes.getEntities().get(0).getMethods().contains("a"));
+        assertEquals("e's method list size is 1", 1, classes.getEntities().get(0).getMethods().size());
+
+        assertFalse("False when creating method that already exists", classes.createMethod("e", "a"));
+        assertFalse("False when creating method that already exists", classes.createMethod("e", "a"));
```

Comment:
This line is tested twice

Reply:
Oh lol. Got rid of the line but I'll wait to see if I need any more changes before committing again.